octodns/provider/ns1.py

```diff
@@ -1020,17 +1128,52 @@ def _params_for_SRV(self, record):
                   for v in record.values]
         return {'answers': values, 'ttl': record.ttl}, None
 
+    def _get_ns1_filters(self, ns1_zone_name):
+        ns1_filters = {}
+        ns1_zone = self._client.zones_retrieve(ns1_zone_name)
+        for ns1_record in ns1_zone['records']:
+            if ns1_record.get('tier', 1) > 1:
+                # Need to get the full record data for geo records
+                full_rec = self._client.records_retrieve(ns1_zone_name,
+                                                         ns1_record['domain'],
+                                                         ns1_record['type'])
+                if 'filters' in full_rec:
+                    filter_key = ns1_record['domain'] + '.'
+                    ns1_filters[filter_key] = full_rec['filters']
+        return ns1_filters
+
+    def _disabled_flag_in_filters(self, filters, domain):
+        disabled_count = ['disabled' in f for f in filters].count(True)
+        if disabled_count and disabled_count != len(filters):
+            # Some filters have the disabled flag, and some don't. Disallow
+            exception_msg = 'Mixed disabled flag in filters for {}'.format(
+                            domain)
+            raise Ns1Exception(exception_msg)
+        return disabled_count == len(filters)
+
     def _extra_changes(self, desired, changes, **kwargs):
         self.log.debug('_extra_changes: desired=%s', desired.name)
-
+        ns1_filters = self._get_ns1_filters(desired.name[:-1])
         changed = set([c.record for c in changes])
-
         extra = []
         for record in desired.records:
             if record in changed or not getattr(record, 'dynamic', False):
                 # Already changed, or no dynamic , no need to check it
                 continue
 
+            # Filter normalization
+            # Check if filters for existing domains need an update
+            # Needs an explicit check since there might be no change in the
+            # config at all. Filters however might still need an update
+            domain = '.'.join([record.name, record.zone.name])
```

Comment:
Seems cleaner to just do a format here rather than building an array and joining.

```suggestion
            domain = '{}.{}'format(record.name, record.zone.name)
```